### PR TITLE
Adding PGP signing to the release process

### PR DESCRIPTION
**Pull request by mattfarina**
**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by technosophos — APPROVED** [review]

**Review by adamreese — APPROVED** [review]

File: docs/release_checklist.md [hunk 1]
```diff
@@ -286,14 +300,14 @@ The community keeps growing, and we'd love to see you there!
...
+- [Linux arm64](https://storage.googleapis.com/kubernetes-helm/helm-vX.Y.Z-linux-arm64.tar.gz) ([checksum](https://storage.googleapis.com/kubernetes-helm/helm-vX.Y.Z-linux-arm64.tar.gz.sha256) / CHECKSUM_VAL)
+- [Linux i386](https://storage.googleapis.com/kubernetes-helm/helm-vX.Y.Z-linux-386.tar.gz) ([checksum](https://storage.googleapis.com/kubernetes-helm/helm-vX.Y.Z-linux-386.tar.gz.sha256) / CHECKSUM_VAL)
+- [Linux ppc64le](https://storage.googleapis.com/kubernetes-helm/helm-vX.Y.Z-linux-ppc64le.tar.gz) ([checksum](https://storage.googleapis.com/kubernetes-helm/helm-vX.Y.Z-linux-ppc64le.tar.gz.sha256) / CHECKSUM_VAL)
+- [Linux s390x](https://storage.googleapis.com/kubernetes-helm/helm-vX.Y.Z-linux-s390x.tar.gz) ([checksum](https://storage.googleapis.com/kubernetes-helm/helm-vX.Y.Z-linux-s390x.tar.gz.sha256) / CHECKSUM_VAL)
+- [Windows amd64](https://storage.googleapis.com/kubernetes-helm/helm-vX.Y.Z-windows-amd64.zip) ([checksum](https://storage.googleapis.com/kubernetes-helm/helm-vX.Y.Z-windows-amd64.zip.sha256) / CHECKSUM_VAL)
```
Comment (bacongobbler) [inline docs/release_checklist.md]: Would you mind explaining what CHECKSUM_VAL is intended to be?

**Review by bacongobbler — COMMENTED** [review]
looks good to me! One comment about CHECKSUM_VAL, otherwise this seems like a good process to follow for the time being.

I wish we could sign the release assets during the CI process, but I understand storing private keys in Circle may be a security risk. Do you think it's worth generating a "Helm CI" shared PGP key that signs release assets right in the CI pipeline, or would that be an anti-pattern for signing releases?

File: docs/release_checklist.md (hunk 1 above)
Comment (mattfarina) [reply docs/release_checklist.md]: That is where we insert the actual checksum for each of these. In addition to the linked file it's displayed here. The reason for this is that we currently have the checksum on the same host/domain as the files. If someone can compromise the file they can compromise the checksum, too. This provides storing the checksum in a second location.

Our plan for notary in v3 will make this a temporary thing for v2, I think.

**Comment by mattfarina** [comment]
@bacongobbler When it comes to signing keys open source projects usually have the people control them. Companies will tend to have signing processes that are tightly controlled so they can highly control their keys. I don't think we should put our signing keys into another companies CI.

The apache foundation has written up a bunch on signing releases for anyone who wants to read about it over at https://www.apache.org/dev/release-signing

**Pull request merged**